File: TD/TD4/Code/2.cpp
/* Effectuer la permutation de trois variables */
#include <iostream>

/* Procédure */

void permut(int &a, int &b, int &c)
{
  int temp;
  temp = b;
  b = a;
  a = c;
  c = temp;
}

/* Utilisation */

int main()
{
  int a;
  int b;
  int c;
  
  a = 5;
  b = 8;
  c = 2;
  
  permut(a, b, c);
  
  std::cout << "a = " << a << std::endl;
  std::cout << "b = " << b << std::endl;
  std::cout << "c = " << c << std::endl;
  
  return 0;
}

File: TD/TD4/Code/Python/2.py
#!/usr/bin/python
# Effectuer la permutation de trois variables


def permute(foo, bar, buz):
    foo, bar, buz = buz, foo, bar
    return foo, bar, buz


foo = 5
bar = 8
buz = 2

foo, bar, buz = permute(foo, bar, buz)  # I ❤ Python
